usr.bin/minimal_count_stats/minimal_count_stats.c

```diff
@@ -28,65 +31,113 @@
 #include <err.h>
 #include <getopt.h>
 #include <inttypes.h>
 #include <libgen.h>
 #include <spawn.h>
 #include <statcounters.h>
 #include <stdbool.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <stdio.h>
-#include <sysexits.h>
+#include <time.h>
 
 extern char **environ;
 
+// clang-format off
 static const struct option options[] = {
 	{ "archname", required_argument, NULL, 'a' },
+	{ "basic", no_argument, NULL, 'b' },
 	{ "csv-noheader", no_argument, NULL, 'c' },
 	{ "help", no_argument, NULL, 'h' },
 	{ "format", required_argument, NULL, 'f' },
 	{ "output", required_argument, NULL, 'o' },
 	{ "progname", required_argument, NULL, 'p' },
 	{ "quiet", no_argument, NULL, 'q' },
+	{ "time-only", no_argument, NULL, 't' },
 	{ "verbose", no_argument, NULL, 'v' },
 	{ NULL, 0, NULL, 0 }
 };
+// clang-format on
+
+static void
+read_counters(statcounters_bank_t *c, bool basic_only, bool time_only)
+{
+	memset(c, 0, sizeof(*c));
+	if (time_only)
+		return;
+
+	if (basic_only) {
+		c->cycles = statcounters_read_cycles();
+		c->instructions = statcounters_read_instructions();
+	} else {
+		statcounters_sample(c);
+	}
+}
 
 static void
 usage(int exitcode)
 {
-	warnx("usage: beri_count_stats [-q/--quiet] [-v/--verbose] [-o file] <command>");
+	warnx("usage: %s [-f format] [-o file] <command>", getprogname());
 	fprintf(stderr, "options:\n");
 	for (int i = 0; options[i].name != NULL; i++) {
-		fprintf(stderr, "  --%s/-%c\n", options[i].name, options[i].val);
+		fprintf(stderr, "  --%s/-%c\n", options[i].name,
+		    options[i].val);
 	}
 
 	exit(exitcode);
 }
 
+static void
+print_basic_human_readable_diff(FILE *f, struct timespec *elapsed_time,
+    statcounters_bank_t *diff)
+{
+	fprintf(f, "Elapsed time:          %jd.%09jds\n",
+	    (intmax_t)elapsed_time->tv_sec, (intmax_t)elapsed_time->tv_nsec);
+	fprintf(f, "cyles:                 %12" PRId64 "\n", diff->cycles);
+	fprintf(f, "instructions:          %12" PRId64 "\n",
+	    diff->instructions);
+}
+
 int
 main(int argc, char **argv)
 {
 	int status;
 	pid_t pid;
 	bool verbose = false;
 	bool quiet = false;
-	const char* output_filename = NULL;
-	const char* progname = NULL;
-	const char* architecture = "unknown";
+	bool basic_only = false;
+	bool time_only = false;
+	const char *output_filename = NULL;
+	const char *progname = NULL;
+	const char *architecture = NULL;
 	int opt;
-	statcounters_fmt_flag_t statcounters_format = (statcounters_fmt_flag_t)-1;
+	statcounters_bank_t start_count;
+	statcounters_bank_t end_count;
+	statcounters_bank_t diff_count;
+	struct timespec start_ts;
+	struct timespec end_ts;
+	struct timespec ts_diff;
+	statcounters_fmt_flag_t statcounters_format =
+	    (statcounters_fmt_flag_t)-1;
 
 	/* Start option string with + to avoid parsing after first non-option */
-	while ((opt = getopt_long(argc, argv, "+a:chf:o:p:qv", options, NULL)) != -1) {
+	while ((opt = getopt_long(argc, argv, "+a:bchf:o:p:qtv", options,
+		    NULL)) != -1) {
```

Comment:
```suggestion
	    NULL)) != -1) {
```
I believe is correct

Reply:
Yeah that's clang format indenting one level for each opening paren.

Reply:
Unfortunately there's no way to turn it off. Maybe I should fix that upstream so I won't have to make style changes on future PRs...